images/hook/entrypoint.sh

```diff
@@ -1,44 +1,66 @@
+    do
+        echo "---> Deleting resources in $namespace namespace"
+        echo "---> Deleting old 'heapster' resources"
+        rig delete deployments/heapster --resource-namespace=$namespace --force
 
-    echo "---> Deleting old 'influxdb' resources"
-    rig delete deployments/influxdb --resource-namespace=kube-system --force
-    rig delete rc/influxdb --resource-namespace=kube-system --force # in case we're upgrading from version where it was still rc
+        echo "---> Deleting old 'influxdb' resources"
+        rig delete deployments/influxdb --resource-namespace=$namespace --force
 
-    echo "---> Deleting old 'grafana' resources"
-    rig delete deployments/grafana --resource-namespace=kube-system --force
-    rig delete rc/grafana --resource-namespace=kube-system --force # in case we're upgrading from version where it was still rc
+        echo "---> Deleting old 'grafana' resources"
+        rig delete deployments/grafana --resource-namespace=$namespace --force
 
-    echo "---> Deleting old deployment 'telegraf'"
-    rig delete deployments/telegraf --resource-namespace=kube-system --force
+        echo "---> Deleting old 'telegraf' resources"
+        rig delete deployments/telegraf --resource-namespace=$namespace --force
+        rig delete daemonsets/telegraf-node --resource-namespace=$namespace --force
 
-    echo "---> Deleting old deployment 'kapacitor'"
-    rig delete deployments/kapacitor --resource-namespace=kube-system --force
+        echo "---> Deleting old deployment 'kapacitor'"
+        rig delete deployments/kapacitor --resource-namespace=$namespace --force
 
-    echo "---> Deleting old configmap 'grafana-cfg'"
-    rig delete configmaps/grafana-cfg --resource-namespace=kube-system --force
-    echo "---> Deleting old configmap 'grafana'"
-    rig delete configmaps/grafana --resource-namespace=kube-system --force
+        echo "---> Deleting old secrets"
+        rig delete secrets/grafana --resource-namespace=$namespace --force
+        rig delete secrets/grafana-influxdb-creds --resource-namespace=$namespace --force
+
+        echo "---> Deleting old configmaps"
+        for cfm in influxdb grafana-cfg grafana grafana-dashboards-cfg grafana-dashboards grafana-datasources
+        do
+            rig delete configmaps/$cfm --resource-namespace=$namespace --force
+        done
+    done
 
-    echo "---> Deleting old secret 'grafana'"
-    rig delete secrets/grafana --resource-namespace=kube-system --force
+    echo  "---> Moving smpt-cofiguration secret to monitoring namespace"
+    if ! /opt/bin/kubectl --namespace=monitoring get secret smtp-configuration > /dev/null 2>&1; then
+        if /opt/bin/kubectl --namespace=kube-system get secret smtp-configuration > /dev/null 2>&1; then
+            /opt/bin/kubectl --namespace=kube-system get secret smtp-configuration --export=true -o json | \
+                jq '.metadata.namespace = "monitoring"' | /opt/bin/kubectl create -f -
```

Comment:
Specifically, in that case, it does not matter. We should check the existence of secret in the new namespace and only if it does not exist move to that namespace. kubectl create will not fail, because there is no secret with the same name in the new namespace.

Reply:
I know it won't fail, I meant for the rollback procedure to work correctly. If you just use kubectl create then the secret will not be removed during rollback, will it?